packages/geo/src/quad.key.ts

```diff
@@ -9,4 +25,90 @@ export const QuadKey = {
         const shortestLength = Math.min(qkA.length, qkB.length);
         return qkA.substr(0, shortestLength) == qkB.substr(0, shortestLength);
     },
+
+    /**
+     * Get the quadkey's children
+     * @example
+     * '3' -> ['30', '31', '32', '33']
+     *
+     * @param qk
+     */
+    children(qk: string): string[] {
+        return ['0', '1', '2', '3'].map((c) => qk + c);
+    },
+
+    /**
+     * Get the parent quadkey for the quadkey
+     *
+     * @example
+     * `31` -> `3`
+     * @param qk
+     */
+    parent(qk: string): string {
+        return qk.substr(0, qk.length - 1);
+    },
+
+    /**
+     * Find duplicate quadkeys and remove them while simplifying groupings
+     *
+     * TODO is there a faster way to do this
+     *
+     * @param quadKeys
+     */
```

Comment:
Faster would be to use a trie instead of a Set. It would automatically remove duplicates

Reply:
Just done some benchmarking it can process 500,000 quadkeys of length 10-15 in < 100ms so it seems fast enough for my usecase.